src/models/Settings.js

```diff
@@ -238,6 +241,57 @@ function (Okta, Q, Errors, BrowserFeatures, Util, Logger, config) {
           return !!(socialAuthConfigured && idpDisplay.toUpperCase() === 'PRIMARY');
         },
         cache: true
+      },
+      configuredIdpsOptionsSet: {
+        deps: ['configuredSocialIdps'],
+        fn: function (configuredSocialIdps) {
+          _.each(configuredSocialIdps, function(idp) {
```

Comment:
In general, we shouldn't be modifying the object they pass in - low risk, but it's possible they are doing something with it as well and changing the properties on the object could affect the developer's code.

```javascript
return _.map(configuredSocialIdps, function (idp) {
  var type = idp.type.toLowerCase();
  return {
    // the properties + idp properties
    // alternatively, clone the idp and then add these properties to it
  };
});
```

And a nit - no need to use `idp['dataAttr']`, use `idp.dataAttr`